storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/get/PreparedGetObject.java

```diff
@@ -0,0 +1,184 @@
+
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.WorkerThread;
+
+import com.pushtorefresh.storio.StorIOException;
+import com.pushtorefresh.storio.contentresolver.ContentResolverTypeMapping;
+import com.pushtorefresh.storio.contentresolver.StorIOContentResolver;
+import com.pushtorefresh.storio.contentresolver.queries.Query;
+
+import rx.Observable;
+
+import static com.pushtorefresh.storio.internal.Checks.checkNotNull;
+
+/**
+ * Represents Get Operation for {@link StorIOContentResolver}
+ * which performs query that retrieves single object
+ * from {@link android.content.ContentProvider}.
+ *
+ * @param <T> type of result.
+ */
+public final class PreparedGetObject<T> extends PreparedGet<T> {
+
+    @NonNull
+    private final Class<T> type;
+
+    @Nullable
+    private final GetResolver<T> explicitGetResolver;
+
+    PreparedGetObject(@NonNull StorIOContentResolver storIOContentResolver,
+                      @NonNull Class<T> type,
+                      @NonNull Query query,
+                      @Nullable GetResolver<T> explicitGetResolver) {
+        super(storIOContentResolver, query);
+        this.type = type;
+        this.explicitGetResolver = explicitGetResolver;
+    }
+
+    /**
+     * Executes Prepared Operation immediately in current thread.
+     * <p>
+     * Notice: This is blocking I/O operation that should not be executed on the Main Thread,
+     * it can cause ANR (Activity Not Responding dialog), block the UI and drop animations frames.
+     * So please, call this method on some background thread. See {@link WorkerThread}.
+     *
+     *  @return single instance of mapped result. Can be {@code null}, if no items are found.
+     */
+    @SuppressWarnings({"ConstantConditions", "NullableProblems"})
+    @WorkerThread
+    @Nullable
+    @Override
+    public T executeAsBlocking() {
+        try {
+            final GetResolver<T> getResolver;
+
+            if (explicitGetResolver != null) {
+                getResolver = explicitGetResolver;
+            } else {
+                final ContentResolverTypeMapping<T> typeMapping = storIOContentResolver.internal().typeMapping(type);
+
+                if (typeMapping == null) {
+                    throw new IllegalStateException("This type does not have type mapping: " +
+                            "type = " + type + "," +
+                            "ContentProvider was not touched by this operation, please add type mapping for this type");
+                }
+
+                getResolver = typeMapping.getResolver();
+            }
+
+            final Cursor cursor = getResolver.performGet(storIOContentResolver, query);
+
+            try {
+                final int count = cursor.getCount();
+
+                if (count == 0) {
+                    return null;
+                }
+
+                cursor.moveToNext();
```

Comment:
`moveToFirst()`, just in case of broken initial position of `Cursor` returned from user's `GetResolver`